src/coin-gecko/api.ts

```diff
@@ -40,3 +41,26 @@ export function useGetRosePrice() {
     },
   )
 }
+
+export type TokenPriceInfo = {
+  price?: number
+  usedGecko?: boolean
+}
+
+export const useTokenPrice = (ticker: NativeTicker): TokenPriceInfo => {
+  const rosePrice = useGetRosePrice().data
```

Comment:
(small note: this makes a request even if picker isnt ROSE)

Reply:
> (small note: this makes a request even if picker isnt ROSE)

Yes, that's because of the stupid rules of hooks. Is there a good way to make the call conditional? Please note that (conditionally) setting `enabled: false` on the query is _not_ sufficient to skip it.